src/isl/isl_constraint.rs

```diff
@@ -171,6 +177,17 @@ impl IslConstraint {
                 value,
                 RangeType::NonNegativeInteger,
             )?)),
+            "element" => {
+                if value.ion_type() != IonType::Symbol && value.ion_type() != IonType::Struct {
+                    return Err(invalid_schema_error_raw(format!(
+                        "element constraint was a {:?} instead of a symbol/struct",
+                        value.ion_type()
+                    )));
+                }
```

Comment:
question—This seems like it should be handled elsewhere. Doesn't the `from_ion_element` function validate its own input?

Reply:
Yes, This is already checked within `from_ion_element` and can be removed from here.